File: daily/面试题18. 删除链表的节点.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var deleteNode = function(head, val) {
    if(head.val==val)
        return head.next;
    var p=head;
    while(p && p.next)
    {
        if(p.next.val==val)
            p.next=p.next.next;
        p=p.next;
    }
    return head;
};

File: daily/C++版本/面试题18. 删除链表的节点.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteNode(ListNode* head, int val) {
        if(head->val==val)
            return head->next;
        ListNode *p=head,*temp;
        while(p && p->next)
        {
            if(p->next->val==val)
                p->next=p->next->next;            
            p=p->next;
        }
        return head;
    }   
};
